### PR TITLE
Adding git hash to the api-reference doc

### PR DESCRIPTION
**Pull request by lukas-bednar**
Fixes: kubevirt-incubator/api-reference#5

kubevirt-api-reference-version

### CODE REVIEW

File: hack/gen-swagger-doc/gen-swagger-docs.sh [hunk 1]
```diff
@@ -63,6 +63,9 @@ if [ "$OUTPUT_FORMAT" = "html" ]; then
     # $$ has special meaning in asciidoc, we need to escape it
     sed -i 's|\$\$|+++$$+++|g' "$WORKDIR/definitions.adoc"
     sed -i '1 i\:last-update-label!:' "$WORKDIR/"*.adoc
+    sed -i -e "/KubeVirt API\$/a\\:revnumber: $(git rev-parse HEAD || echo no-revision)" \
+            -e "/__Terms of service__ :/a\\__Version__ : {revnumber}" \
```
Comment (rmohr) [inline hack/gen-swagger-doc/gen-swagger-docs.sh]: Maybe we can make it even a link to the core repo?

Comment (lukas-bednar) [reply hack/gen-swagger-doc/gen-swagger-docs.sh]: Done

**Comment by lukas-bednar** [comment]
@fabiand yes, it is good idea.
In addition I want to add links to generated releases into [api-reference/README.md](https://github.com/kubevirt-incubator/api-reference/blob/master/README.md) automatically.
Because right now there is link for master only.

**Comment by rmohr** [comment]
> In addition I want to add links to generated releases into api-reference/README.md automatically.
> Because right now there is link for master only.

That is exactly what I was about to comment. I would make that simple and generate the whole README.md within the deploy script.

**Comment by rmohr** [comment]
> EDIT: Generating does obviously not work ... Updating it will work.

Have to correct me again. We can generate it ased on the folders.

File: hack/gen-swagger-doc/deploy.sh [hunk 2]
```diff
@@ -5,24 +5,44 @@ set -e
...
+# Generate README.md file
+cat > README.md << __EOF__
+# KubeVirt API Reference
+
+Content of this repository is generated from OpenAPI specification of
+[KubeVirt project](https://github.com/kubevirt/kubevirt) .
+
+## KubeVirt API References
+
+* [master](${GITHUB_IO_FQDN}/index.html)
```
Comment (rmohr) [inline hack/gen-swagger-doc/deploy.sh]: Shouldn't the master documentation be in a `master` sub-folder because of `$TARGET_DIR`?

Comment (lukas-bednar) [reply hack/gen-swagger-doc/deploy.sh]: I did rebase on latest changes and resolved conflicts.
Yes you are right, there is missing `master` sub directory, thanks.

File: hack/gen-swagger-doc/deploy.sh [hunk 3]
```diff
@@ -5,24 +5,44 @@ set -e
...
+# Generate README.md file
+cat > README.md << __EOF__
+# KubeVirt API Reference
+
+Content of this repository is generated from OpenAPI specification of
+[KubeVirt project](https://github.com/kubevirt/kubevirt) .
+
+## KubeVirt API References
+
+* [master](${GITHUB_IO_FQDN}/master/index.html)
```
Comment (rmohr) [inline hack/gen-swagger-doc/deploy.sh]: One last question: would we get two master links if the master directory then exists already?

Comment (lukas-bednar) [reply hack/gen-swagger-doc/deploy.sh]: I think we will not, there is regex `"^v[0-9.]*"` which matches directories starting with `v`.
But I am open to other suggestions to make less error prone!

Comment (rmohr) [reply hack/gen-swagger-doc/deploy.sh]: no change necessary. works for me. Let's see what happens.

File: .travis.yml [hunk 4]
```diff
@@ -69,6 +69,7 @@ deploy:
   skip_cleanup: true
   script: bash hack/gen-swagger-doc/deploy.sh
   on:
+    tags: true
```
Comment (lukas-bednar) [inline .travis.yml]: it doesn't get executed now for non-tagged commits. I assumed that it will ... now I read doc and it doesn't :-(

**Pull request merged**